.github/workflows/dev_on_workflow_tester.yml

```
@@ -107,6 +130,80 @@ jobs:
           restore-keys: |
             v1-coverage-dependency-cache-${{ runner.os }}-${{ github.sha }}-${{ matrix.shard }}
           key: v1-coverage-dependency-cache-${{ runner.os }}-${{ github.sha }}-${{ matrix.shard }}
+
+      - name: Cache the allure reports
+        id: allure-report-cache-native
+        if: github.ref == 'refs/heads/master'
+        uses: pass-culture-github-actions/gcs-cache@v1.0.0
+        with:
+          bucket: ${{ inputs.CACHE_BUCKET_NAME }}
+          path: |
+            allure-results-${{matrix.shard}}
+          restore-keys: |
+            v1-allure-dependency-cache-${{ runner.os }}-${{ github.sha }}-${{ matrix.shard }}
+          key: v1-allure-dependency-cache-${{ runner.os }}-${{ github.sha }}-${{ matrix.shard }}
+
+  allure-report:
+    name: Update Allure report
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-22.04
+    permissions:
+      id-token: write
+      contents: read
+    needs: yarn_test_native
+
+    steps:
+      - uses: actions/checkout@v4.2.1
+      - name: 'Authentification to Google'
+        uses: 'google-github-actions/auth@v2'
+        with:
+          workload_identity_provider: ${{ secrets.GCP_EHP_WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ secrets.GCP_EHP_SERVICE_ACCOUNT }}
+
+      - name: 'Get secrets (github)'
+        id: 'secrets'
+        uses: 'google-github-actions/get-secretmanager-secrets@v2'
+        with:
+          secrets: |-
+            API_TOKEN_GITHUB:passculture-metier-ehp/passculture-main-sa-access-token
```

pourquoi on utilise un token de main ?

```suggestion
            API_TOKEN_GITHUB:passculture-metier-ehp/passculture-app-native-sa-access-token
```